zebra-consensus/src/block/check.rs

```diff
@@ -131,9 +134,25 @@ pub fn subsidy_is_valid(block: &Block, network: Network) -> Result<(), BlockErro
         // Funding streams are paid from Canopy activation to the second halving
         // Note: Canopy activation is at the first halving on mainnet, but not on testnet
         // ZIP-1014 only applies to mainnet, ZIP-214 contains the specific rules for testnet
-        tracing::trace!("funding stream block subsidy validation is not implemented");
-        // Return ok for now
-        Ok(())
+
+        let funding_streams = subsidy::funding_streams::funding_stream_values(height, network)
+            .expect("We always expect a funding stream hashmap response even if empty");
+
+        let mut counter = 0;
+        for (_receiver, amount) in funding_streams {
+            if !subsidy::general::find_output_with_amount(coinbase, amount).is_empty() {
+                counter += 1;
+            }
+        }
+
+        // If we can find `FUNDING_STREAM_NUMBER_OF_RECEIVERS` matches,
+        // then funding streams amounts in coinbase are valid.
+        // TODO: Validate funding stream addresses
+        if counter == FUNDING_STREAM_NUMBER_OF_RECEIVERS {
+            Ok(())
```

Comment:
What is the consensus rule you are implementing here?

You might want something like `funding_stream_amounts.is_subset(output_amounts)`.
(Since the funding stream amounts are all unique, we can just ignore duplicate amounts and addresses.)
https://doc.rust-lang.org/std/collections/struct.HashSet.html#method.is_subset

But I think the consensus rules also require matching the addresses.
In a future PR, we can easily change `HashSet` code to check `(Address, Amount)` pairs.

Reply:
I think this suggestion is great so i implemented in https://github.com/ZcashFoundation/zebra/pull/3017/commits/ff0b145202dfce1f56c5605a2d08a707e8aafe3f

Also added the consensus rule text we are implementing as a comment.

Reply:
Looks great!